### ISSUE TITLE
KWIC-Suchfeld - colour problem

### ISSUE THREAD

**Issue by Fera333** [dh-network/quadriga#13]
In dem Suchkasten kann man die eingegebenen Wörter nicht sehen, scheinbar ist die Schrift weiß auf weißem Hintergrund. Nicht sicher ob das nur ein Darkmode Problem ist.

**Comment by hsluytergaethje** [comment]
Ja, das ist ein Problem mit Darkmode. Dazu haben wir bisher noch keine Lösung, aber sollten in dem Notebook darauf hinweisen. Beim NLP-Notebook ist es dasselbe Problem.

**Comment by lam1aa** [comment]
Was the problem happening in this searchbox? It seems to be working in both light and dark mode now.
![image](https://github.com/user-attachments/assets/6ec9033e-22bb-474d-8631-aaa984b906f6)

**Comment by hsluytergaethje** [comment]
It's about the word input fields in the jupyter notebooks. See for example the input fields in the chapter "[🚀 Korpusverarbeitung – Annotation mit spaCy](https://dh-network.github.io/quadriga/nlp-enrichment/NLP-Enrichment.html#anzeigen-von-worthaufigkeiten)" or in chapter "[Analyse 2: Keyword in Context (KWIC)](https://dh-network.github.io/quadriga/word_search/FS_1_MVP_Analysis_KWIC.html#extraktion-und-darstellung-der-wortkontexte-durch-kwic)". 

In this screenshot you see that the text only gets visible when highlighted:
<img width="1101" alt="Screenshot 2025-01-08 at 08 42 53" src="https://github.com/user-attachments/assets/e728b55a-a2f6-4aff-96db-306fac30ebeb" />

**Comment by lam1aa** [comment]
I am experiencing two problems while solving this issue in [Analyse 2: Keyword in Context (KWIC)](https://dh-network.github.io/quadriga/word_search/FS_1_MVP_Analysis_KWIC.html#extraktion-und-darstellung-der-wortkontexte-durch-kwic)":

Text Input Box Not Updating Table:
- When adding text to the input field from JupyterBook or Colab, it does not update the table as expected.
- After changing the text I reran the cells downwards (kwic.contextwords..) but it did not change.
- Currently, it only works if the search terms are changed directly in the code, not through the text input box. And user can only do it from the ipynb file, not from the website.

Dark Mode Styling Issue:
- I attempted to fix the Bokeh theme styling for dark mode, but it was not working in any way.
- As an alternative, I tried using ipywidgets, which allows for easily configurable styling.

However, I am unsure of the exact behavior of the text input and how it should function. Could you please look into its current behavior? @hsluytergaethje

**Comment by lam1aa** [comment]
To fix the color problem in the text input box dark mode, we need to change the default style of the Bokeh theme. Since Bokeh widgets (version 3.0+) use the Shadow DOM, modifying the CSS globally from any CSS file won’t work. Instead, we have to add the CSS directly in the notebook. For example:

from bokeh.models import TextInput

```
custom_css = """
:host .bk-input {
    background-color: #f0f0f0; /* Your desired color */
}
"""

text_input = TextInput(
    value='',
    title="Enter text:",
    stylesheets=[custom_css]
)
```

This ensures the style is applied properly, but it needs to be repeated in each notebook that is using `TextInput`. I tested it on the [🚀 Korpusverarbeitung – Annotation mit spaCy](https://dh-network.github.io/quadriga/nlp-enrichment/NLP-Enrichment.html#anzeigen-von-worthaufigkeiten) notebook, and it works. I'll push the changes after fixing some other issues.
